packages/hub/src/components/EntityInfo.tsx

```diff
@@ -1,29 +1,76 @@
-import { FC } from "react";
+import { FC, cloneElement, ReactNode } from "react";
 import Link from "next/link";
 
-import { userRoute } from "@/routes";
+import { clsx } from "clsx";
+import { IconProps } from "@/relative-values/components/ui/icons/Icon";
 
 // works both for models and for definitions
-export const EntityInfo: FC<{
-  username: string;
+export type entityNode = {
   slug: string;
   href: string;
-}> = ({ username, slug, href }) => {
+  icon?: FC<IconProps>;
+};
+
+const ListWithSeparator: React.FC<{
+  items: ReactNode[];
+  separator: (r: number) => ReactNode;
+}> = ({ items, separator }) => {
+  return items.reduce<React.ReactNode[]>((accumulator, item, index) => {
```

Comment:
This is a big gnarly. Code suggested by ChatGPT

Reply:
Yeah, this is weird.
I'd rewrite it with `items.map`, no need for reduce.